### PR TITLE
Upgrade to FFI 1.0

### LINKED ISSUE #86: ffi/primitives is failing on dev channel
This sample should work once `package:ffi` is upgraded to `1.0.0`: https://github.com/dart-lang/samples/pull/87

See [this failed job](https://github.com/dart-lang/samples/pull/85/checks?check_run_id=1811567266#step:6:12) for details

### PR DESCRIPTION
**Pull request by mit-mit**
fixes #86

### CODE REVIEW

**Review by dcharkes — APPROVED** [review]
lgtm

**Comment by mit-mit** [comment]
@dcharkes please review again; I added all the other changes we discussed

**Review by johnpryan — APPROVED** [review]

File: ffi/primitives/primitives.dart [hunk 1]
```diff
@@ -55,7 +55,7 @@ main() {
 
   // calls int subtract(int *a, int b);
   // Create a pointer
-  Pointer<Int32> p = allocate();
+  Pointer<Int32> p = calloc<Int32>();
```
Comment (dcharkes) [inline ffi/primitives/primitives.dart:58]: The type argument should be inferred.

**Review by dcharkes — APPROVED** [review]
lgtm, but the CI fails because of the required Dart version.

File: ffi/primitives/primitives.dart (hunk 1 above)
Comment (mit-mit) [reply ffi/primitives/primitives.dart]: Done

**Comment by mit-mit** [comment]
I'm going to wait a bit to merge this until we get CI passing

**Comment by domesticmouse** [comment]
@mit-mit can you update this PR and resolve conflicts?

**Pull request merged**